test/unit/math/test_ad.hpp

```diff
@@ -333,12 +334,25 @@ template <typename G>
 void expect_ad_derivatives(const ad_tolerances& tols, const G& g,
                            const Eigen::VectorXd& x) {
   double gx = g(x);
-  test_gradient(tols, g, x, gx);
+  if (!tols.gradient_val_.is_inf() || !tols.gradient_grad_.is_inf()) {
```

Comment:
Is there a reason not to call `std::isinf(...)` directly?

Reply:
Yes, because `tol.gradient_val_` isn't a primitive---it's a class, and the `is_inf` function is a method on that class.